docs/_guide/abi-typegen/using-generated-types.md

```diff
@@ -46,7 +48,38 @@ const { value, logs } = await script.functions.main(1).call();
 console.log({ value, logs });
 ```
 
+# Using Generated Predicate Types
+
+After generating types via:
+
+```console
+yarn exec fuels -i ./abis/*-abi.json -o ./types --predicate
+```
+
+We can use these files like so:
+
+```ts
+import { Wallet } from "fuels";
+import { MyPredicate__factory } from "./types";
+
+const wallet = Wallet.fromAddress("...");
+const predicate = MyPredicate__factory.createInstance();
+
+await predicate
+  .setData({ foo: 'bar' })
+  .transfer(wallet.address, <amount>);
+
+const walletBalance = await wallet.getBalance();
+const predicateBalance = await predicate.getBalance();
+
+console.log({
+  walletBalance,
+  predicateBalance,
+});
+```
+
```

Comment:
@camsjams I tried stitching this usage example from the e2es for predicates. Do you think it is helpful the way it is? I would appreciate any suggestions you could give me.

The `MyPredicate__factory.createInstance();` part at `line:66` on the example above refers to the generated factory method, which looks like this:

https://github.com/FuelLabs/fuels-ts/blob/8217c08dce9a7e907ee5db2e7806f74449011c43/packages/abi-typegen/test/fixtures/templates/predicate/factory.hbs#L85-L93

> **Note** _Although not ideal, I left a `TODO` for replacing the plain-text code blocks for imported code snippets, which I want to tackle soon via #730._

Reply:
@camsjams Thank you. https://github.com/FuelLabs/fuels-ts/pull/750/commits/708d9047ee05e37a7a1abb8712570d516a67ae81